src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterSimpleRequestBuilder.java

```diff
@@ -0,0 +1,113 @@
+import static org.mule.extension.http.internal.request.KeyValuePairUtils.toMultiMap;
+import static org.mule.runtime.http.api.server.HttpServerProperties.PRESERVE_HEADER_CASE;
+
+import org.mule.extension.http.internal.request.HttpRequesterConfig;
+import org.mule.extension.http.internal.request.UriUtils;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.extension.api.annotation.param.NullSafe;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.display.DisplayName;
+import org.mule.runtime.extension.api.annotation.param.display.Text;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.request.HttpRequestBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Component that specifies how to create a proper HTTP simple request, that can be used in a Source
+ * and connectivity testing.
+ *
+ * @since 1.7
+ */
+public class HttpRequesterSimpleRequestBuilder {
+
+  /**
+   * The body of the response message
+   */
+  @Parameter
+  @Optional(defaultValue = "")
+  @Text
+  @DisplayName("Body")
+  private String requestBody;
+
+  /**
+   * HTTP headers the message should include.
+   */
+  @Parameter
+  @Optional
+  @NullSafe
+  @DisplayName("Headers")
+  protected List<SimpleRequestHeader> requestHeaders = emptyList();
+
+  /**
+   * URI parameters that should be used to create the request.
+   */
+  @Parameter
+  @Optional
+  @DisplayName("URI Parameters")
+  private Map<String, String> requestUriParams = emptyMap();
+
+  /**
+   * Query parameters the request should include.
+   */
+  @Parameter
+  @Optional
+  @DisplayName("Query Parameters")
+  private List<QueryParam> requestQueryParams = emptyList();
+
+  public TypedValue<Object> getRequestBody() {
+    return TypedValue.of(requestBody);
+  }
+
+  public void setRequestBody(String body) {
+    this.requestBody = body;
+  }
+
+  public List<SimpleRequestHeader> getRequestHeaders() {
+    return unmodifiableList(requestHeaders);
+  }
+
+  public void setRequestHeaders(List<SimpleRequestHeader> headers) {
+    this.requestHeaders = headers != null ? headers : emptyList();
+  }
+
+  public String replaceUriParams(String path) {
+    return UriUtils.replaceUriParams(path, requestUriParams);
+  }
+
+  public List<QueryParam> getRequestQueryParams() {
+    return unmodifiableList(requestQueryParams);
+  }
+
+  public Map<String, String> getRequestUriParams() {
+    return unmodifiableMap(requestUriParams);
+  }
+
+  public void setRequestQueryParams(List<QueryParam> queryParams) {
+    this.requestQueryParams = queryParams;
+  }
+
+  public void setRequestUriParams(Map<String, String> uriParams) {
+    this.requestUriParams = uriParams;
+  }
+
+  public HttpRequestBuilder configure(HttpRequesterConfig config) {
```

Comment:
misleading name. this method gets the impression that `this` instance will get configured when what is actually doing is creating a builder